### PR TITLE
ModuleHelper: added a do_raise() method to MH base class

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Added a method `do_raise()` to MH base class, so that module developers won't have to import ModuleHelperException unless they want to subclass it.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/module_utils/mh/base.py

### CODE REVIEW

File: changelogs/fragments/4660-mh-added-do-raise.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - ModuleHelper module utils - ``ModuleHelperBase` now has a convenience method ``do_raise`` (https://github.com/ansible-collections/community.general/pull/4660).
```
Comment (felixfontein) [inline changelogs/fragments/4660-mh-added-do-raise.yaml:2]:
```suggestion
  - ModuleHelper module utils - ``ModuleHelperBase`` now has a convenience method ``do_raise`` (https://github.com/ansible-collections/community.general/pull/4660).
```

File: plugins/module_utils/mh/base.py [hunk 2]
```diff
@@ -31,6 +31,11 @@ def __init__(self, module=None):
     def diff_mode(self):
         return self.module._diff
 
+    def do_raise(self, *args, **kwargs):
+        exc_class = kwargs.get("exception_class", _MHE)
```
Comment (felixfontein) [inline plugins/module_utils/mh/base.py:35]: Such constructs are a cause for quite a few security vulnerabilities, where user-provided data suddenly controls execution due to some obscure feature of a library. I would stick to the default class here.
```suggestion
        exc_class = _MHE
```

Comment (russoz) [reply plugins/module_utils/mh/base.py:35]: okie

File: plugins/module_utils/mh/base.py [hunk 3]
```diff
@@ -31,6 +31,11 @@ def __init__(self, module=None):
     def diff_mode(self):
         return self.module._diff
 
+    def do_raise(self, *args, **kwargs):
+        exc_class = _MHE
+        kwargs = dict((k, v) for k, v in kwargs.items() if k != 'exception_class')
```
Comment (felixfontein) [inline plugins/module_utils/mh/base.py:36]: Do you intentionally want to keep the `exception_class` filter here?

Comment (russoz) [reply plugins/module_utils/mh/base.py]: No sense at all... I was in a hurry. Further simplified it.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks!

**Pull request merged**